File: examples/java-maven-example/src/test/java/org/galaxio/performance/picatinny/scenarios/TransactionScenario.java
package org.galaxio.performance.picatinny.scenarios;

import io.gatling.javaapi.core.ScenarioBuilder;
import org.galaxio.performance.picatinny.cases.PicatinnyCases;

public final class TransactionScenario {
    private TransactionScenario() {
    }

    public static ScenarioBuilder apply(String name, String transactionName) {
        return PicatinnyScenario.withFeeders(name)
                .exec(PicatinnyCases.businessOperation(transactionName));
    }
}

File: examples/kotlin-gradle-example/src/gatling/kotlin/org/galaxio/performance/picatinny/scenarios/TransactionScenario.kt
package org.galaxio.performance.picatinny.scenarios

import io.gatling.javaapi.core.ScenarioBuilder
import org.galaxio.performance.picatinny.cases.PicatinnyCases

object TransactionScenario {
    fun apply(name: String, transactionName: String): ScenarioBuilder =
        PicatinnyScenario.withFeeders(name)
            .exec(PicatinnyCases.businessOperation(transactionName))
}
